src/__tests__/BoardSizeForm.test.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { shallow, mount } from 'enzyme';
+import BoardSizeForm from '../components/BoardSizeForm';
+
+describe('BoardSizeForm component', () => {
+  it('renders a form', () => {
+    const wrapper = shallow(
+      <BoardSizeForm
+        handleChange={jest.fn()}
+        handleSubmit={jest.fn()}
+        hasError={false}
+      />
+    );
+    expect(wrapper.exists('#size-form')).toBeTruthy();
+  });
+
+  it('renders an error if given one', () => {
+    const wrapper = mount(
+      <BoardSizeForm
+        handleChange={jest.fn()}
+        handleSubmit={jest.fn()}
+        hasError={true}
+      />
+    );
+    expect(
+      wrapper.contains(
+        <p>Error: Invalid input. Please input a number from 3 - 10</p>
+      )
+    ).toBeTruthy();
+  });
+
+  it('calls its handleChange method', () => {
+    const mockFn = jest.fn();
+    const wrapper = shallow(
+      <BoardSizeForm
+        handleChange={mockFn}
+        handleSubmit={jest.fn()}
+        hasError={false}
+      />
+    );
+    expect(mockFn.mock.calls).toHaveLength(0);
+    wrapper
+      .find('#size-input')
```

Comment:
This could also have been `.find('input[type="text"]')`, but I think that adding a `name` attribute here would also be more appropriate than adding an ID. If you add an accompanying `aria-label`, you've got accessibility!

The `.at(0)` stuff is brittle and not declarative, but here's the approach I would take to fixing that:

1. A more specific selector, e.g. `form input[type="text"]`
2. If the form is likely to have more than one text input, that would be a good time to give it a name: `form input[name="boardSize"]`
3. If there can be more than one instance of the component on the page, you'll want to make the selector even more specific: `form.sideBar input[name="boardSize"]`

The point is: __think through the problem, and only add as much as you need__. Point 3 above is an especially important one for react components (reusability), and a great reason to avoid using IDs in your elements.

Reply:
Said it better than I did, haha.